### ISSUE TITLE
Question about attention_mask / position_ids initialization

### ISSUE THREAD

**Issue by RyanMetcalfeInt8** [microsoft/onnxruntime-genai#1682]
Wanted to ask a question here if that's okay. I've been trying to track down some accuracy issues we're encountering on my side when we I ORT GenAI (as compared with OpenVINO GenAI) for certain models. 

I've found that the attention_mask is prepared slightly differently between the two frameworks. In ORT GenAI, it initializes the attention_mask here: https://github.com/microsoft/onnxruntime-genai/blob/main/src/models/position_inputs.cpp#L245

Here, the attention_mask entries are set to 1, except for tokens that are equal to the model's pad_token_id -- in which case they're set to 0.

In my case, the input_id's don't have any padding added by the tokenizer but may have some EOS tokens from chat templating, which happen to be equal to the pad_token_id -- and so they are set to 0 which seems to affect accuracy. 

As far as I understand, the attention_mask should be set to 1 for all tokens except for explicit pad tokens that may added by the tokenizer (i.e. to left/right pad to a particular max length, etc.).

For example, with native pytorch transformers, the tokenizer will generate an attention mask of all 1's.

The same kind of logic is also used for initializing position_ids: https://github.com/microsoft/onnxruntime-genai/blob/main/src/models/position_inputs.cpp#L183

I think it may be a bug, but I wanted to get some feedback on perhaps what the intent is of these initialization functions are. Thanks!

**Comment by kunal-vaishnavi** [comment]
Closing this issue since it has been fixed [here](https://github.com/microsoft/onnxruntime-genai/pull/1694)
